### PR TITLE
Support Remote Config on "Share Dialog" closes #3088

### LINKED ISSUE #3088: Support Remote Config on "Share Dialog"
**Why/User Benefit/User Problem**
To make something easier ...etc
To be completed 

**User story**
As a Mozilla, we would like to remote config the content of the "Share Dialog" to A/B test the conversion rate and observe if we can increase "referral" installs 

**Acceptance Criteria (how do I know when I’m done?)** 
1. Being able to remote config the share dialog message
2. Being able to remote config the share link of the share dialog
3. Being able to remote config the share message of the share dialog

**Additional context**

### put correct labels ###
Add corresponding labels from the right panel to make it clearer
telemetry wanted - does this introduce adding or modifying telemetry?
UX/VD wanted - does this require UX/VD input to make the story ready for development
l10n wanted - does this introduce adding or modifying in-product strings?
help wanted - does this feature suitable for people other than the core team to wok on?

### PR DESCRIPTION
**Pull request by cnevinc**
I can't find a method to add a new line character in Firebase remote config.
After discussing with @elin-moco 
we'll use `<BR>` in Firebase console  Remote Config setting and replace it with `\n` 
Developers should work with marketing team if they want to do this replacement in the code base.

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java [hunk 1]
```diff
@@ -283,6 +285,16 @@ void refreshRemoteConfigDefault(Context context) {
         map.put(FirebaseHelper.SCREENSHOT_CATEGORY_MANIFEST, ScreenshotManager.SCREENSHOT_CATEGORY_MANIFEST_DEFAULT);
         map.put(FirebaseHelper.VPN_RECOMMENDER_PACKAGE, FeatureSurveyViewHelper.Constants.PACKAGE_RECOMMEND_VPN);
         map.put(FirebaseHelper.VPN_RECOMMENDER_URL, FeatureSurveyViewHelper.Constants.LINK_RECOMMEND_VPN);
+        // Share App
+        map.put(FirebaseHelper.SHARE_APP_DIALOG_TITLE, context.getString(R.string.rate_app_dialog_text_title,
+                context.getString(R.string.app_name)));
+        // we used \n as new line, but the caller side now required HTML <br> as new line
+        map.put(FirebaseHelper.SHARE_APP_DIALOG_CONTENT,
+                context.getString(R.string.share_app_dialog_text_content).replace("\n", "<br>"));
```
Comment (cnevinc) [inline app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java]: this is ugly. Maybe we should re-translate it?

**Comment by cnevinc** [comment]
confirmed with @elin-moco , we'll use <BR> in remote config, only make it `\n` before using in TextView. This will only work for text in the dialog's content TextView

File: app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java [hunk 2]
```diff
@@ -276,6 +278,15 @@ void refreshRemoteConfigDefault(Context context) {
             map.put(FirebaseHelper.RATE_APP_DIALOG_TEXT_POSITIVE, context.getString(R.string.rate_app_dialog_btn_go_rate));
             map.put(FirebaseHelper.RATE_APP_DIALOG_TEXT_NEGATIVE, context.getString(R.string.rate_app_dialog_btn_feedback));
             map.put(FirebaseHelper.FIRST_LAUNCH_NOTIFICATION_MESSAGE, context.getString(R.string.preference_default_browser) + "?\uD83D\uDE0A");
+            // Share App
+            map.put(FirebaseHelper.STR_SHARE_APP_DIALOG_TITLE, context.getString(R.string.rate_app_dialog_text_title,
```
Comment (mTwTm) [inline app/src/main/java/org/mozilla/focus/utils/FirebaseHelper.java]: rate_app_dialog_text_title should be share_app_dialog_textview_title

**Comment by Daisy-pliu** [comment]
Verify: ✅
Test version : 1.1.4 (10425).Nightly
Test step:
1. open [Remote config in Firebase](https://console.firebase.google.com/u/0/project/rocketnightly/config)
2. set str_share_app_dialog_title=title <a href="http://google.com">link</a>
3. set str_share_app_dialog_content= 1😊\n 2\uD83D\uDE0A\n 3 \u000a😊
4. set str_share_app_dialog_msg = I'm a happy url <a href="http://google.com">link</a>
5. publish

Test result:
1. share dialog
screenshot_20190304-165218
2. intents with text 
screenshot_20190304-165246

**Pull request merged**